### DISCUSSION TITLE
Override method

### DISCUSSION THREAD

**Discussion by Vliro** [mfussenegger/nvim-jdtls#164] [General]
Hi! I am really enjoying working on Java using nvim-jdtls. However, one functionality I cannot seem to find is the simple override method, that is list all methods of the superclasses/interfaces and generate an empty body. Is this possible in nvim-jdtls? Thanks :)

**Comment by mfussenegger** [comment]
For methods that must be implemented you can use a code action on the class level. For optional methods you can use completion with snippet support. (So you need a plugin like nvim-cmp or nvim-lsp-compl)

Here is both in action:

https://user-images.githubusercontent.com/38700/139428022-40ce20bd-3699-4461-be68-bf5044269b33.mp4

**Reply by Vliro** [reply]
Thank you @mfussenegger . I know about the mandatory one for e.g. interfaces but not the autocomplete, that looks great. Is it somehow possible to get a list of methods, and select one? For instance, via telescope to mimic other editors. Thanks!

**Reply by mfussenegger** [reply]
If you trigger the completion within the class body without having typed anything it should list all methods, but it will also include keywords and so on.

I guess you could use the lower-level API to filter these out. I have something like that in my dotfiles but instead of filtering for methods it filters for dedicated snippets triggering:

https://github.com/mfussenegger/dotfiles/blob/7989acbadc29fc0ab85f7358be8571eee67d9fff/vim/.config/nvim/lua/me/snippet.lua#L20-L63

So you'd use either `vim.lsp.buf_request_sync` or `vim.lsp.buf_request` on the `textDocument/completion` LSP-method and filter the results to fit your use-case and pass the result to telescope
